hr_expense_operating_unit/tests/test_hr_expense_operating_unit.py

```diff
@@ -0,0 +1,185 @@
+        expense_sheet.approve_expense_sheets()
+        expense_sheet.action_sheet_move_create()
+        self.assertEqual(
+            expense_sheet.account_move_id.line_ids[0].operating_unit_id,
+            expense_sheet.operating_unit_id
+        )
+
+    def test_security(self):
+        # User 2 is only assigned to Operating Unit B2C, and cannot
+        # Access Expenses of Main Operating Unit.
+        record = self.hr_expense_model.sudo(self.user2.id).search(
+            [('id', '=', self.hr_expense1.id),
+             ('operating_unit_id', '=', self.ou1.id)])
+        self.assertEqual(record.ids, [], 'User 2 should not have access to %s'
+                         % self.ou1.name)
+        # Create the expense sheet
+        hr_expense_dict1 = self.hr_expense1.action_submit_expenses()
+        self.hr_expense_sheet1 = self.hr_expense_sheet_model.create({
+            'name': hr_expense_dict1['context']
+            ['default_name'],
+            'employee_id': hr_expense_dict1['context']
+            ['default_employee_id'],
+            'expense_line_ids': [(4,
+                                  hr_expense_dict1['context']
+                                  ['default_expense_line_ids'][0])],
+            'operating_unit_id': hr_expense_dict1['context']
+            ['default_operating_unit_id']
+        })
+        self.hr_expense1.write({
+            'sheet_id': self.hr_expense_sheet1.id
+        })
+        self._post_journal_entries(self.hr_expense_sheet1)
+        # Expense OU should have same OU of its accounting entries
+        self.assertEqual(
+            self.hr_expense_sheet1.expense_line_ids.operating_unit_id,
+            self.hr_expense_sheet1.account_move_id.line_ids.mapped(
+                'operating_unit_id'),
+            "Expense OU should match with accounting entries OU")
+
+    def test_constrains_error(self):
+        with self.assertRaises(ValidationError):
+            hr_expense_dict1 = self.hr_expense1.action_submit_expenses()
+            self.hr_expense_sheet1 = self.hr_expense_sheet_model.create({
+                'name': hr_expense_dict1['context']
+                ['default_name'],
+                'employee_id': hr_expense_dict1['context']
+                ['default_employee_id'],
+                'expense_line_ids': [(4,
+                                      hr_expense_dict1['context']
+                                      ['default_expense_line_ids'][0])],
+                'operating_unit_id': hr_expense_dict1['context']
+                ['default_operating_unit_id']
+
+            })
+            self.hr_expense1.write({
+                'sheet_id': self.hr_expense_sheet1.id
+            })
+            self.hr_expense_sheet1.expense_line_ids.write({
+                'operating_unit_id': self.b2c.id
+            })
+
+        company_id = self.env['res.company'].create({
+            'name': 'My Company',
+            'partner_id': self.partner1.id,
+            'rml_header1': 'My Company',
```

Comment:
@bjeficent @AaronHForgeFlow @HviorForgeFlow No need to `rml_header1`

Reply:
Thanks for the notice. Could you create a PR with the fix?